src/gennylib/include/gennylib/PhaseLoop.hpp

```diff
@@ -77,10 +78,44 @@ class IterationCompletionCheck final {
         }
     }
 
-    explicit IterationCompletionCheck(PhaseContext& phaseContext)
-        : IterationCompletionCheck(phaseContext.get<TimeSpec, false>("Duration"),
-                                   phaseContext.get<UIntSpec, false>("Repeat"),
-                                   phaseContext.isNop()) {}
+    explicit IterationChecker(PhaseContext& phaseContext)
+        : IterationChecker(phaseContext.get<TimeSpec, false>("Duration"),
+                           phaseContext.get<IntegerSpec, false>("Repeat"),
+                           phaseContext.isNop()) {
+        auto rateSpec = phaseContext.get<RateSpec, false>("Rate");
```

Comment:
it would be a mis-configuration to have `Rate` but not `Repeat`/`Duration`. Can we check that and bomb out if specified `Rate` but not `Repeat` or `Duration`?

Reply:
Ah good point! I misunderstood your previous question about rate limited ops potentially running in the wrong phase.